Document/0x07b-Testing-Network-Communication.md

```diff
@@ -227,3 +227,87 @@ Ensure that critical operations require at least one additional channel to confi
 * [1] The Mobile Application Security Verification Standard - https://github.com/OWASP/owasp-masvs/blob/master/Document/0x03-Using_the_MASVS.md
 * [2] Infobip 2FA library - https://2-fa.github.io/libraries/android-library.html
 * [3] Google Authenticator for Android - https://github.com/google/google-authenticator-android
+
+
+### Testing SSL/TLS Pinning
+
+#### Overview
+
+Certificate pinning allows to hard-code the certificate or parts of it into the app that is known to be used by the server<sup>[1]</sup>. This technique is used to reduce the threat of a rogue CA and CA compromise. Pinning the server’s certificate takes the CA out of the game. Mobile apps that implement certificate pinning only can connect to a limited numbers of servers, as a small list of trusted CAs or server certificates are hard-coded in the application.
+
+Implementing certificate pinning is not a trivial task. Most applications will not benefit from using this technique and may introduce new vulnerabilities if they implement it incorrectly or without regard to operational challenges<sup>1</sup>:
+
+- It does not protect against compromises of the pinned certificate itself.
+- It makes the operations more complex and thus requires a more mature process.
+- It may break SSL/TLS validation.
+- It can be circumvented by reverse engineering or on rooted/jailbroken devices.
+
+Even large organizations with mature processes and large devops staff are not immune to mistakes in implementing certificate pinning<sup>[2]</sup>. One high profile case involving a bank involved an expired intermediate certificate and caused outage to the customers.
+
+Once the challenges are understood and the proper processes are being implemented, important decisions have to be made:
+
+- Which level in the certificate chain should be pinned, CA, intermediate, or leaf? Pinning to the most specific certificate reduces flexibility, pinning to the CA increases vulnerability.
```

Comment:
> pinning to the CA increases vulnerability

Is this correct? It is possible that you are totally secure if you pin to the CA. It increases the risk in some cases. Maybe this can be simplified to "Pinning to the most specific certificate reduces flexibility, but increases security."